src/uicomponents/PokemonLookup.tsx

```diff
@@ -0,0 +1,1238 @@
+    paddingRight: 0,
+    borderBottom: 'solid #aaaaaa 2px',
+}));
+
+const RoundedRow = styled(TableRow)(({ theme }) => ({
+    marginTop: 10,
+    marginBottom: 10,
+    paddingTop: 10,
+    paddingBottom: 10,
+    height: "50px",
+    border: 10,
+    backgroundColor: theme.palette.mode === 'light'
+      ? lighten(theme.palette.background.paper, 0.6)
+      : darken(theme.palette.background.paper, 0.2),
+}));
+
+const ButtonRow = styled(RoundedRow)(({ theme }) => ({
+    '&:hover': {
+        backgroundColor: theme.palette.action.hover,
+    },
+    '&:click': {
+        backgrtoundColor: theme.palette.action.selected,
+    }
+}));
+
+const HeaderRow = styled(RoundedRow)(({ theme }) => ({
+    backgroundColor: theme.palette.background.paper,
+}));
+
+const HeaderButton = styled(Button)(({ theme }) => ({
+    minWidth: 0,
+    color: theme.palette.text.primary,
+}));
+
+const CompactTableCell = styled(TableCell)(({ theme }) => ({
+    padding: 0,
+    paddingLeft: 1,
+    paddingRight: 1,
+}));
+
+const CompactLeftCell = styled(CompactTableCell)(({ theme }) => ({
+    borderRadius: "100px 0px 0px 100px",
+    paddingLeft: 10,
+}));
+
+const CompactRightCell = styled(CompactTableCell)(({ theme }) => ({
+    borderRadius: "0px 100px 100px 0px",
+    paddingRight: 10,
+}));
+
+function SortingButton({label, method, sortMethod, sortDirection, handleButtonClick, translationKey}: {label: string, method: string, sortMethod: string, sortDirection: string, handleButtonClick: (label: string) => void, translationKey: any}) {
+    const sortIcon =    method !== sortMethod ? <UnfoldMoreIcon sx={{ m: 0}} /> :
+                        sortDirection === "desc" ? <KeyboardArrowDown sx={{ m: 0}}/> :
+                        <KeyboardArrowUpIcon sx={{ m: 0}}/>;
+
+    return (
+        <HeaderButton 
+            size="small" 
+            variant="text" 
+            onClick={() => handleButtonClick(sortMethod)} 
+            style={{ backgroundColor: 'transparent' }}
+            sx={{ textTransform: 'none', minWidth: 0 }}
+        >
+            <Typography fontSize={10} fontWeight={method === sortMethod ? 800 : 600}>
+                {`${getTranslation(label, translationKey,"stats")}`}
+            </Typography>
+            { sortIcon }
+        </HeaderButton>
+    )
+
+}
+
+function SpeciesSearchResult({pokemon, allSpecies, handleSetPokemon, translationKey}: {pokemon: SpeciesName, allSpecies: Map<SpeciesName,PokemonData>, handleSetPokemon: (s: SetOption) => void, translationKey: any}) {
+    const [open, setOpen] = useState(false);
+    if (pokemon.includes("Flab")) { // Something odd might be happening with the unicode representaiton of é somewhere
+        pokemon = "Flabebe" as SpeciesName;
+    }
+    const data = allSpecies.get(pokemon)!;
+    const sets = raiderSetMap.get(pokemon) || 
+    [{
+        name: "Blank Set",
+        pokemon: pokemon,
+        level: 100,
+        nature: "Hardy",
+        ability: data.abilities[0].name,
```

Comment:
Crashed the app. Unfortunately I dont know how to reproduce the issue but the issue is reading from undefined here.

Reply:
Seems to happen if you're searching a filter but don't select anything, then close and reopen the modal

Reply:
Looks like it might have to do with how items in the GroupedVirtuoso are rendered. I logged `pokemon` there being move names...

But rather than worry about it too much, we can just clear the input when the modal is closed. It seems to fix the problem.